lib/oca-epak/epak/client.rb

```diff
@@ -100,22 +177,30 @@ def get_operation_codes
       # @param [String] "From date" in DD-MM-YYYY format
       # @param [String] "To date" in DD-MM-YYYY format
       # @return [Array, nil] Contains an array of hashes with NroProducto and NumeroEnvio
-      def list_shipments(cuit, from_date, to_date)
+      def list_shipments(opts = {})
+        seconds_in_a_day = 86400
+        since_date = opts[:since_date] || (Time.now - 7 * seconds_in_a_day).strftime("%d-%m-%Y")
+        until_date = opts[:until_date] || Time.now.strftime("%d-%m-%Y")
+
         method = :list_envios
-        opts = { "CUIT" => cuit, "FechaDesde" => from_date,
-                 "FechaHasta" => to_date }
-        response = client.call(method, message: opts)
-        parse_results_table(response, method)
+        message = {
+          "CUIT" => opts[:cuit],
+          "FechaDesde" => since_date,
+          "FechaHasta" => until_date
+        }
+
+        response = client.call(method, message: message)
+        parse_result(response, method)
       end
 
       # Returns all provinces in Argentina
       #
       # @return [Array, nil] Provinces in Argentina with their ID and name as a Hash
       def provinces
-        response = client.call(:get_provincias)
-        if body = response.body[:get_provincias_response]
-          body[:get_provincias_result][:provincias][:provincia]
-        end
+        method = :get_provincias
```

Comment:
@cavi21 I don't see the need for a local variable here. You can just use `:get_provincias` below and avoid a new variable.